testing/databaseTest.py

```diff
@@ -970,6 +976,20 @@ def kinetics_check_reactant_and_product_template(self, family_name):
             if boo:
                 raise ValueError("Error occured in databaseTest. Please check log warnings for all error messages.")
 
+    def kinetics_check_num_reactant_and_product(self, family_name):
+        """
+        This test checks that if the number of reactants and products are specified in the groups.py for
+        rate rules that are generated from the ATG.
+        """
+        family = self.database.kinetics.families[family_name]
+        if family.auto_generated:
+            if not getattr(family, 'reactant_num'):
+                logging.error(f'The number of reactants is not defined in the family {family_name}')
+            if not getattr(family, 'reactant_num'):
+                logging.error(f'The number of products is not defined in the family {family_name}')
+            if not getattr(family, 'reactant_num') or not getattr(family, 'reactant_num'):
```

Comment:
Should the second conditional say `getattr(family, 'product_num')` ?

Reply:
My mistake. Thanks for catching that

Reply:
@xiaoruiDong Can you push a change that updates this line before we merge?

Reply:
Thanks! I am too careless about this.